src/Microsoft.Identity.Web/TokenCacheProviders/Distributed/MsalDistributedTokenCacheAdapter.cs

```diff
@@ -82,6 +82,12 @@ protected override async Task RemoveKeyAsync(string cacheKey)
             catch (Exception ex)
             {
                 _logger.LogError($"[IdWebCache] Connection issue encountered with Distributed cache. Currently using In Memory cache only. Error message: {ex.Message} ");
+
+                if (_distributedCacheOptions.OnL2CacheFailure != null && _distributedCacheOptions.OnL2CacheFailure(ex))
+                {
+                    _logger.LogDebug($"[IdWebCache] DistributedCache: Retry to remove cacheKey {cacheKey}. ");
+                    await _distributedCache.RemoveAsync(cacheKey).ConfigureAwait(false);
```

Comment:
>await _distributedCache.RemoveAsync(cacheKey).ConfigureAwait(false); [](start = 20, length = 68)

have to catch this one as well

Reply:
yes. fixed.

---
In reply to: [587807756](https://github.com/AzureAD/microsoft-identity-web/pull/1046#discussion_r587807756) [](ancestors = 587807756)